Baremetal http issue on openebs-operator

# Bug report

## Bug description
openebs-operator can't reach k8s' api server because it tries to use a `https` scheme instead of the `http` defined clusterwide.
**What you expected to happen**:
maya doesn't have this issue, so I expected to be the same on operator.

## Bug Reproduction 
Setup an up to date (`1.9.1`) k8s bare metal cluster. Use the firewall security model described here : https://kubernetes.io/docs/getting-started-guides/scratch/#security-models
Run the pod.

# Else
Slack related archive begins here : https://openebs-community.slack.com/archives/C3NPGQ6G3/p1517394852000413 and ends here : https://openebs-community.slack.com/archives/C3NPGQ6G3/p1517405899000094

@theonlydoo, I think I have got the issue reproduced using minikube. I am using the following test yaml to check if the fix is going to work. Can you please apply the following yaml in your setup and share the output. 

*Please replace 10.128.0.12 with your kubernetes server IP address and 8080 with the insecure port.*
```
apiVersion: v1
kind: Pod
metadata:
  name: k8s-go-client
spec:
  containers:
  - image: openebs/k8s-client:latest
    imagePullPolicy: Always
    name: k8s-go-client
    args :
    -  /go/src/github.com/openebs/test-storage/k8s-client/listpods.go
    - -master=10.128.0.12:8080
```

Here is the information/output of running the above test file in my minikube (using localkube) setup where apiserver is listening only on insecure port 8080:

```
sudo netstat -nap | grep localkube | grep LISTEN | grep -v LISTENING

Output:
tcp        0      0 127.0.0.1:10248         0.0.0.0:*               LISTEN      17901/localkube 
tcp        0      0 127.0.0.1:2379          0.0.0.0:*               LISTEN      17901/localkube 
tcp        0      0 127.0.0.1:2380          0.0.0.0:*               LISTEN      17901/localkube 
tcp6       0      0 :::4194                 :::*                    LISTEN      17901/localkube 
tcp6       0      0 :::10250                :::*                    LISTEN      17901/localkube 
tcp6       0      0 :::10251                :::*                    LISTEN      17901/localkube 
tcp6       0      0 :::10252                :::*                    LISTEN      17901/localkube 
tcp6       0      0 :::10255                :::*                    LISTEN      17901/localkube 
tcp6       0      0 :::30000                :::*                    LISTEN      17901/localkube 
tcp6       0      0 :::8080                 :::*                    LISTEN      17901/localkube 
```

The API server is available at `10.128.0.12:8080`

```
kubectl --server='10.128.0.12:8080' get nodes

Output:
NAME             STATUS    ROLES     AGE       VERSION
kmova-minikube   Ready     <none>    5h        v1.9.0
```

In this minikube setup, I am running the following test.yaml:
```
apiVersion: v1
kind: Pod
metadata:
  name: k8s-go-client
spec:
  containers:
  - image: openebs/k8s-client:latest
    imagePullPolicy: Always
    name: k8s-go-client
    args :
    -  /go/src/github.com/openebs/test-storage/k8s-client/listpods.go
    - -master=10.128.0.12:8080
```
Note the way IP address and port are specified. 

```
kubectl --server='10.128.0.12:8080' apply -f test.yaml

Output
pod "k8s-go-client" created
```

The above command launches a pod that compiles client-go code and fetches the list of pods and pvs. It may take 1/2 mins before the output starts to appear:

```
kubectl --server='10.128.0.12:8080' logs k8s-go-client

Output:
Either master or kubeconfig specified. building kube config from that..
There are 7 pods in the cluster
There are 0 PVs in the cluster
There are 7 pods in the cluster
There are 0 PVs in the cluster
There are 7 pods in the cluster
There are 0 PVs in the cluster
There are 7 pods in the cluster
.
.
.
```

If I don't specify the master address as below:

```
apiVersion: v1
kind: Pod
metadata:
  name: k8s-go-client-incluster
spec:
  containers:
  - image: openebs/k8s-client:latest
    imagePullPolicy: Always
    name: k8s-go-client
    #replace the listpods.go in the following args to your desired test file
    args :
    -  /go/src/github.com/openebs/test-storage/k8s-client/listpods.go
```

I see the following in the logs:
```
kubectl --server='10.128.0.12:8080' logs -f k8s-go-client-incluster

Ouptut:
panic: Get https://10.96.0.1:443/api/v1/persistentvolumes: dial tcp 10.96.0.1:443: getsockopt: connection refused

goroutine 1 [running]:
panic(0xe84920, 0xc42028e310)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
main.main()
        /go/src/github.com/openebs/test-storage/k8s-client/listpods.go:63 +0x3cf
exit status 2
```

--- a/openebs/openebs-provisioner.go
+++ b/openebs/openebs-provisioner.go
@@ -240,7 +240,7 @@ func main() {
 	} else {
 		// Create an InClusterConfig and use it to create a client for the controller
 		// to use to communicate with Kubernetes
-		config, err := rest.InClusterConfig()
+		config, err = rest.InClusterConfig()
 	}
 	if err != nil {
 		glog.Errorf("Failed to create config: %v", err)

--- a/openebs/openebs-provisioner.go
+++ b/openebs/openebs-provisioner.go
@@ -35,6 +35,12 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	master     = flag.String("master", "", "Master's URL to communicate with kubernetes-master if running from outside the cluster or if apiserver is allowing insecure/Non SSL connections.")
+	kubeconfig = flag.String("kubeconfig", "", "Absolute path to kubeconfig if running from outside the cluster.")
 )
 
 const (
@@ -224,10 +230,18 @@ func main() {
 
 	flag.Parse()
 	flag.Set("logtostderr", "true")
-
-	// Create an InClusterConfig and use it to create a client for the controller
-	// to use to communicate with Kubernetes
-	config, err := rest.InClusterConfig()
+	var (
+		config *rest.Config
+		err    error
+	)
+	if *master != "" || *kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags(*master, *kubeconfig)
+		fmt.Printf("Client config was built using flags: Address: '%s' Kubeconfig: '%s' \n", *master, *kubeconfig)
+	} else {
+		// Create an InClusterConfig and use it to create a client for the controller
+		// to use to communicate with Kubernetes
+		config, err := rest.InClusterConfig()
+	}
 	if err != nil {
 		glog.Errorf("Failed to create config: %v", err)
 	}

